src/constraints.jl

```diff
@@ -18,9 +18,9 @@ supported in specific circumstances, for example, `F`-in-`S` constraints cannot
 combined with another type of constraint, it should still return `true`.
 """
 function supports_constraint(
-    ::ModelLike,
-    ::Type{<:AbstractFunction},
-    ::Type{<:AbstractSet},
+    @nospecialize(::ModelLike),
+    @nospecialize(F::Type{<:AbstractFunction}),
```

Comment:
Since it's a fallback returning `false`, I don't believe it will affect the performance of bottleneck to looks good

Reply:
Maybe add a comment explaining this